Framework/Algorithms/test/ResampleXTest.h

```diff
@@ -161,6 +170,258 @@ class ResampleXTest : public CxxTest::TestSuite {
     // Remove workspace from the data service.
     AnalysisDataService::Instance().remove(outWSName);
   }
+
+  // Set-up a generic function for running tests for EventWorkspace
+  void do_testResampleXEventWorkspace(EventType eventType, bool inPlace,
+                                      bool PreserveEvents,
+                                      bool expectOutputEvent) {
+
+    // Two events per bin
+    EventWorkspace_sptr test_in =
+        WorkspaceCreationHelper::createEventWorkspace2(2, 100);
+    test_in->switchEventType(eventType);
+
+    std::string inName("test_inEvent");
+    std::string outName("test_inEvent_output");
+    if (inPlace)
+      outName = inName;
+
+    AnalysisDataService::Instance().addOrReplace(inName, test_in);
+    ResampleX alg;
+    alg.initialize();
+    alg.setPropertyValue("InputWorkspace", inName);
+    alg.setPropertyValue("OutputWorkspace", outName);
+    alg.setPropertyValue("Xmin", "0.0,0.0");
+    alg.setPropertyValue("Xmax", "100,150");
+    alg.setPropertyValue("NumberBins", "50");
+    alg.setProperty("PreserveEvents", PreserveEvents);
+    TS_ASSERT(alg.execute());
+    TS_ASSERT(alg.isExecuted());
+
+    MatrixWorkspace_sptr outWS;
+    EventWorkspace_sptr eventOutWS;
+    TS_ASSERT_THROWS_NOTHING(
+        outWS = boost::dynamic_pointer_cast<MatrixWorkspace>(
+            AnalysisDataService::Instance().retrieve(outName)));
+    TS_ASSERT(outWS);
+    if (!outWS)
+      return;
+
+    // Is the output gonna be events?
+    if (expectOutputEvent) {
+      eventOutWS = boost::dynamic_pointer_cast<EventWorkspace>(outWS);
+      TS_ASSERT(eventOutWS);
+      if (!eventOutWS)
+        return;
+      TS_ASSERT_EQUALS(eventOutWS->getNumberEvents(), 2 * 100 * 2);
+      // Check that it is the same workspace
+      if (inPlace)
+        TS_ASSERT(eventOutWS == test_in);
+    }
+
+    // Define tolerance for ASSERT_DELTA
+    double tolerance = 1.0e-10;
+
+    // Check the first workspace index (from 0 to 100)
+    auto &X1 = outWS->x(0);
+    auto &Y1 = outWS->y(0);
+    auto &E1 = outWS->e(0);
+
+    TS_ASSERT_EQUALS(X1.size(), 51);
+    TS_ASSERT_DELTA(X1[0], 0.0, tolerance);
+    TS_ASSERT_DELTA(X1[1], 2.0, tolerance);
+    TS_ASSERT_DELTA(X1[2], 4.0, tolerance);
+
+    TS_ASSERT_EQUALS(Y1.size(), 50);
+    TS_ASSERT_DELTA(Y1[0], 4.0, tolerance);
+    TS_ASSERT_DELTA(Y1[1], 4.0, tolerance);
+    TS_ASSERT_DELTA(Y1[2], 4.0, tolerance);
+
+    TS_ASSERT_EQUALS(E1.size(), 50);
+    TS_ASSERT_DELTA(E1[0], sqrt(4.0), tolerance);
+    TS_ASSERT_DELTA(E1[1], sqrt(4.0), tolerance);
+
+    // Check the second workspace index (from 0 to 150)
+    auto &X2 = outWS->x(1);
+    auto &Y2 = outWS->y(1);
+    auto &E2 = outWS->e(1);
```

Comment:
Personally I would usually attempt to test as much of the data as possible, i.e., either make the workspace smaller (just 2 spectra with 2 bins each), or write the asserts in a loop, maybe.